src/overview/reducer.js

```diff
@@ -28,6 +28,11 @@ const defaultState = {
     },
     showFilter: false,
     showOnlyBookmarks: false,
+    pageIdForTag: '',
+    newTag: '',
+    resultTags: [],
```

Comment:
All this state and complexity around mutating it could be reduced if you expand on this `resultTags` array. You could maybe have it as an array of custom shaped objects representing the state of each tag in the UI. A tag itself is really simple: it has a selected state + display value, so maybe `resultTags` could be shaped like:

```js
[
  { isSelected: true, value: 'test' },
  { isSelected: true, value: 'education' },
  { isSelected: false, value: 'other tag' },
  { isSelected: true, value: 'good' },
]
```

Each click on the tag in the UI could just update the `isSelected` flag for the related tag state, rather than actually sending a req to bg script to remove it straight away. 

When user presses "Done", you could just compare the original state (call `fetchTagsForPage`) with the current `resultsTag` state and determine the needed `delTags` and `addTags` requests. Then you wouldn't need `deleteTags` state. It may be useful to have a `setTags` method where you can just pass a page ID + any tags and overwrite the old entry? If you think this is better, let me know and I can write it.

`newTag` state should be replaceable with a state for the `<input>`'s value. The `pageIdForTag` state seems weird, but I think it is needed for now.

Reply:
I think instead of `addTags` and `delTags` the `setTags` will be the best, so that we can just pass the new value no matter add or delete, we can use in just one function.
So you can write it and I'll do change accordingly.

Reply:
`setTags` method should be available now if you rebase on upstream (overwrite any conflicts with upstream changes). Also renamed `fetchTagsForPage` to just `fetchTags` to be consistent with other tags bg script methods, so make sure to update the 2 `remoteFunction` instantiations.

Reply:
Seeing how you're accessing the tabs state, it's probably going to be a lot easier for you to store the state as an dictionary of keys to objects. Like:
```js
{
  'test': { isSelected: true, value: 'test' },
  'education': { isSelected: true, value: 'education' },
  'other tag': { isSelected: false, value: 'other tag' },
  'good': { isSelected: true, value: 'good' },
}
```

You can do this as we can assume all tags are unique by their `value`. Then you won't need all the `findIndex` calls everywhere, you can just grab tags by value. You can use lodash's `keyBy` function to auto convert array to object via specified key (or you can just use `Array.prototype.reduce` to reduce the array into object)